js/src/provider/WalletLinkProvider.ts

```diff
@@ -214,16 +214,69 @@ export class WalletLinkProvider
     }
   }
 
-  private async switchEthereumChain(rpcUrl: string, chainId: number) {
+  private async addEthereumChain(
+    chainId: number,
```

Comment:
What's the breakoff point when we change constructor to be objects instead of params?

Reply:
seems like something eslint should tell us, no?